plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/flexible/FlexDeployPreferencesPanel.java

```diff
@@ -58,33 +67,50 @@ protected void createCenterArea() {
         includeOptionalConfigurationFilesButton, "includeOptionalConfigurationFiles");
   }
 
-  private void createAppYamlSection() {
+  /**
+   * Helper method to create a row of a {@link Label}, a {@link Text} for a file path input, and
+   * a "browse" {@link Button} to open a {@link FileDialog}. It also does data binding between
+   * the {@link Text} field and the given {@code modelPropertyName}.
+   */
+  protected Text createBrowseFileRow(String modelPropertyName, String labelText, String Tooltip,
```

Comment:
Would it make sense to provide the databinding property rather than the `modelPropertyName`?  Isolate assumptions that the model is a POJO?

Reply:
Thanks, I could think of better refactoring, which is consistent with an existing pattern.